Include/Rover/Autobox.hpp

```diff
@@ -25,6 +27,25 @@ namespace Rover {
   */
   template<typename T>
   decltype(auto) make_autobox(T&& value);
+
+  template<typename T>
+  struct autobox<T, std::enable_if_t<is_object_pointer_v<T>>> {
```

Comment:
This will treat all pointers as a ```Box<U>```, but only pointers to generators should be boxed. You'll need some kind of type trait defined to test if a type represents a generator, something similar to the following:

```std::declval<T>().generate(std::declval<Evaluator>())```

Reply:
You also need to handle boxing non-generators, ie. autobox_t<int> == Constant<int>.

Reply:
Yep I thought about that, but I was not sure if we were handling promotion to Constant in autobox this time - otherwise testing against T being generator / pointer to generator is not necessary.
As for the Box itself - does it still only accept generators and generator pointers?

Reply:
Box's constructor is strictly for generators and pointers yes. That will never change. Non-generators get boxed as ```Constant```s, even non-generator pointers ie. ```auto_box_t<int*> == Constant<int*>```.